### PR TITLE
TFLM: Remove uint8 support for CMSIS-NN kernels

### PR DESCRIPTION
**Pull request by mansnils**
Some CMSIS-NN kernels are still having uint8 support.

* Remove uint8 support for remaining CMSIS-NN kernels.
* Remove uint8 support for equivalent reference kernels.
* Remove uint8 unit tests for equivalent kernels.
* Replace uint8 default model in network tester to a int8 model.

This is a fix for: https://github.com/tensorflow/tensorflow/issues/48757

### CODE REVIEW

**Comment by gbaned** [comment]
@mansnils Can you please resolve conflicts? Thanks!

**Comment by mansnils** [comment]
@njeffrie ping for review

**Comment by gbaned** [comment]
@mansnils  Can you please resolve conflicts? Thanks!

**Comment by mansnils** [comment]
@njeffrie ping for review

File: tensorflow/lite/micro/examples/network_tester/input_data.h [hunk 1]
```diff
@@ -16,8 +16,93 @@ limitations under the License.
 #ifndef TENSORFLOW_LITE_MICRO_EXAMPLES_NETWORK_TESTER_INPUT_DATA_H_
 #define TENSORFLOW_LITE_MICRO_EXAMPLES_NETWORK_TESTER_INPUT_DATA_H_
 
-static const int input_data_len = 16;
-static const unsigned char input_data[1][16] = {
-    {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}};
+static const int input_data_len = 1024;
+static const unsigned char input_data[] = {
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/network_tester/input_data.h:20]: is this change part of the current PR?

File: tensorflow/lite/micro/examples/network_tester/network_model.h [hunk 2]
```diff
@@ -17,54 +17,150 @@ limitations under the License.
...
-    0x65, 0x64, 0x2e, 0x00, 0x03, 0x00, 0x00, 0x00, 0x18, 0x00, 0x00, 0x00,
-    0x0c, 0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0xf8, 0xff, 0xff, 0xff,
-    0xfc, 0xff, 0xff, 0xff, 0x04, 0x00, 0x04, 0x00, 0x04, 0x00, 0x00, 0x00,
-    0xf8, 0xfe, 0xff, 0xff, 0x20, 0x00, 0x00, 0x00, 0x14, 0x00, 0x00, 0x00,
-    0x08, 0x00, 0x00, 0x00, 0x3c, 0x01, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
-    0x01, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x02, 0x00, 0x00, 0x00, 0x08, 0x00, 0x00, 0x00, 0x94, 0x00, 0x00, 0x00,
-    0x7e, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x03, 0x10, 0x00, 0x00, 0x00,
-    0x02, 0x00, 0x00, 0x00, 0x1c, 0x00, 0x00, 0x00, 0x30, 0x00, 0x00, 0x00,
+    0x1c, 0x00, 0x00, 0x00, 0x54, 0x46, 0x4c, 0x33, 0x00, 0x00, 0x12, 0x00,
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/network_tester/network_model.h:20]: is this change part of the current PR?

**Review by advaitjain — CHANGES_REQUESTED** [review]
Small clarification around why network_tester needs to change in this PR. Other than that, looks good to me.

File: tensorflow/lite/micro/examples/network_tester/input_data.h (hunk 1 above)
Comment (mansnils) [reply tensorflow/lite/micro/examples/network_tester/input_data.h:20]: Yes the original model is a uint8 one. So we need an int8 model instead. It is changed from a uint8 maxpool operator to an int8 depthwise operator.

Comment (mansnils) [reply tensorflow/lite/micro/examples/network_tester/input_data.h:20]: So therefore I needed to change the default input data.

File: tensorflow/lite/micro/examples/network_tester/network_model.h (hunk 2 above)
Comment (mansnils) [reply tensorflow/lite/micro/examples/network_tester/network_model.h:20]: Yes it is changed to an int8 version.

**Review by advaitjain — APPROVED** [review]

File: tensorflow/lite/micro/examples/network_tester/network_model.h (hunk 2 above)
Comment (freddan80) [reply tensorflow/lite/micro/examples/network_tester/network_model.h:20]: I believe the previous test network was uint8.

**Comment by advaitjain** [comment]
Unfortunately, this PR had to be reverted because it (unexpectedly) made the detection_postprocess kernel fail on bluepill.

Here is a PR that reverts the revert and we can recreate the failure: https://github.com/tensorflow/tensorflow/pull/49485

Its not clear to me why the CI passed for the current PR since it is failing for me 100% of the time.

**Comment by advaitjain** [comment]
Also, when you make a new PR, can you also remove
https://github.com/tensorflow/tensorflow/blob/5177e09c9f302372137f28ab7798a8a413c352f7/tensorflow/lite/micro/tools/make/Makefile#L445

**Pull request merged**